### PR TITLE
Allow to disable warning dialog for unsaved files when linting

### LINKED ISSUE #101: Don't lint unsaved files
Currently the extension tries to lint unsaved files but it can't, so it shows a notification every time that says "please save", which is really annoying. Could we please get an option to turn this off?

### PR DESCRIPTION
**Pull request by apupier**
doNotShowDaialogAgain

fixes #101

### CODE REVIEW

**Review by jdkato — APPROVED** [review]

**Comment by ChrisChinchilla** [comment]
@apupier and @jdkato I've been travelling with quite poor internet, will check tomorrow and also bring back the quick actions functionality and we can then make a new release.

File: package.json [hunk 1]
```diff
@@ -60,7 +60,12 @@
 						"Sets `minAlertLevel` to `error`, overriding any configuration files."
 					],
 					"markdownDescription": "Defines from which level of errors and above to display in the problems output."
-				}
+				},
+        "vale.doNotShowAgainWarningForFileToBeSavedBeforeLinting": {
```
Comment (ChrisChinchilla) [inline package.json:64]:
```suggestion
        "vale.doNotShowWarningForFileToBeSavedBeforeLinting": {
```

File: package.json [hunk 2]
```diff
@@ -60,7 +60,12 @@
...
+          "type": "boolean",
+          "default": false,
+          "description": "Do not show warning dialog that a file must be saved to have linting."
```
Comment (ChrisChinchilla) [inline package.json:67]:
```suggestion
          "description": "Do not show warning dialog that a file must be saved to be linted."
```

File: src/features/vsUtils.ts [hunk 3]
```diff
@@ -82,9 +82,22 @@ export const toTitle = (alert: IValeErrorJSON, suggestion: string): string => {
  * @param document The document to check
  * @return Whether the document is elligible
  */
-export const isElligibleDocument = (document: vscode.TextDocument): boolean => {
+export const isElligibleDocument = async (document: vscode.TextDocument): Promise<boolean> => {
   if (document.isDirty) {
-    vscode.window.showErrorMessage("Please save the file before linting.");
+    const config = vscode.workspace.getConfiguration('vale');
+    const shouldIgnore = config.get<boolean>('doNotShowAgainWarningForFileToBeSavedBeforeLinting') === true;
```
Comment (ChrisChinchilla) [inline src/features/vsUtils.ts:88]:
```suggestion
    const shouldIgnore = config.get<boolean>('doNotShowWarningForFileToBeSavedBeforeLinting') === true;
```

File: src/features/vsUtils.ts [hunk 4]
```diff
@@ -82,9 +82,22 @@ export const toTitle = (alert: IValeErrorJSON, suggestion: string): string => {
...
+      return false;
+    }
+
+    const abortAndDoNotShowAgain = 'Abort and do not show again this warning';
+    const choice = await vscode.window.showErrorMessage("Please save the file before linting.",
+      "Abort",
+      abortAndDoNotShowAgain
+    );
+    if (choice === abortAndDoNotShowAgain) {
+      await config.update('doNotShowAgainWarningForFileToBeSavedBeforeLinting', true, true);
```
Comment (ChrisChinchilla) [inline src/features/vsUtils.ts:99]:
```suggestion
      await config.update('doNotShowWarningForFileToBeSavedBeforeLinting', true, true);
```

**Review by ChrisChinchilla — CHANGES_REQUESTED** [review]
Made small suggestions on changing the variable to something clearer. I also cannot replicate the issue that made the need for this fix, but it doesn't seem to affect anything negatively, so, I guess it's fine to include :)

**Pull request merged**